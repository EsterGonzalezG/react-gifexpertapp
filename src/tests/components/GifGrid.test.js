import { shallow } from 'enzyme';
import React from 'react';
import { GifGrid } from './../../components/GifGrid';
import { useFetchGifs } from './../../hooks/useFetchGifs';
jest.mock('./../../hooks/useFetchGifs');

describe('GifGrid component', () => {
  const category = 'One Punch';
  test('should be render component', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should be items when img useFetchGifs', () => {
    const imgs = [
      {
        id: 'ans',
        url: 'https//localhost/kskdask.jpg',
        title: 'title',
      },
      {
        id: 'kaskskask',
        url: 'https//localhost/kskdask.jpg',
        title: 'title',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: imgs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(imgs.length);
  });
});
