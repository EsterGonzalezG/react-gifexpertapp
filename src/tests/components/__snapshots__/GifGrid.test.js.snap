// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid component should be render component 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
